### PR TITLE
Modernize embulk-input-file with v0.11-ready API/SPI

### PR DESCRIPTION
**Pull request by dmikurube**

### CODE REVIEW

File: embulk-input-file/build.gradle [hunk 1]
```diff
@@ -62,6 +63,92 @@ javadocJar {
...
+                        email = "muga.nishizawa@gmail.com"
+                    }
+                    developer {
+                        name = "Dai MIKURUBE"
+                        email = "dmikurube@treasure-data.com"
+                    }
+                    developer {
+                        name = "Koji Arai"
+                        email = "jca02266@gmail.com"
+                    }
```
Comment (dmikurube) [inline embulk-input-file/build.gradle:115]: @jca02266 We're going to separate `embulk-standards` plugins as individual plugins, then to release them as independent pure-Java Embulk plugin JARs into Maven Central. In this `embulk-input-file`, we wanted to include you as a developer from the pull request #1022. What do you think?

File: embulk-input-file/build.gradle [hunk 2]
```diff
@@ -62,6 +63,92 @@ javadocJar {
...
+
+                developers {
+                    developer {
+                        name = "Sadayuki Furuhashi"
+                        email = "frsyuki@gmail.com"
+                    }
+                    developer {
+                        name = "Hiroyuki Sato"
+                        email = "hiroysato@gmail.com"
+                    }
```
Comment (dmikurube) [inline embulk-input-file/build.gradle:103]: @hiroyuki-sato  We're going to separate `embulk-standards` plugins as individual plugins, then to release them as independent pure-Java Embulk plugin JARs into Maven Central. In this `embulk-input-file`, we wanted to include you as a developer from the pull request #585. What do you think?

File: embulk-input-file/build.gradle (hunk 1 above)
Comment (jca02266) [reply embulk-input-file/build.gradle:115]: No problem. Thank you for letting me know.

File: embulk-input-file/build.gradle (hunk 2 above)
Comment (hiroyuki-sato) [reply embulk-input-file/build.gradle:103]: Thank you! No problem.

**Comment by dmikurube** [comment]
`embulk-util-file` will be updated to `0.1.2` in another PR after this is merged to `master`. (#1410 needed to be on this branch to use `0.1.2`, but this PR was developed before #1410. Rebasing everything would be hard...)

**Comment by dmikurube** [comment]
`embulk-util-file:0.1.3` would work. Updated.

**Review by huylenq — APPROVED** [review]
LGTM

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**